deps/src/main/java/com/microsoft/azure/sdk/iot/deps/util/Base64.java

```diff
@@ -283,6 +286,36 @@ public static String encodeBase64StringLocal(byte[] dataValues) throws IllegalAr
         return new String(encodeBase64Internal(dataValues));
     }
 
+    /**
+     * Delegates base64 decoding to apache commons
+     * @param base64Values the values to decode
+     * @return the decoded values
+     */
+    public static byte[] decodeBase64(final byte[] base64Values)
+    {
+        return org.apache.commons.codec.binary.Base64.decodeBase64(base64Values);
```

Comment:
I saw someone's complaining Android support issue with apache codec. Please double check with android API 19 device.

Reply:
I believe that is only versions of apache codec greater than 1.2 that are missing from Android

https://stackoverflow.com/questions/2047706/apache-commons-codec-with-android-could-not-find-method

We should definitely wait to check this code in until we have a lower version of Android at the gate though